From 2075ac59f23beaaa1fba012e4318f50a97cd429f Mon Sep 17 00:00:00 2001
From: Bryce Poole <Bryce.Poole@intel.com>
Date: Mon, 14 Mar 2016 09:00:19 -0700
Subject: [PATCH] mmc: Setting the minimum frequency for puma7

The minimum frequency for puma7 eMMC controllers is 200kHz. If 100kHz
is attempted, the host controller loses the HW interrupt.
---
 drivers/mmc/host/sdhci-acpi.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/mmc/host/sdhci-acpi.c b/drivers/mmc/host/sdhci-acpi.c
index 8f427bf..6b0c010 100644
--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -128,6 +128,12 @@ void cgm_write_b(struct sdhci_host *host, u8 val, int reg)
 		writeb(val, host->ioaddr + reg);
 }
 
+unsigned int sdhci_acpi_cgm_get_min_clock(struct sdhci_host *host)
+{
+	/* CGM host controllers only support down to 200kHz */
+	return 200000;
+}
+
 static const struct sdhci_ops sdhci_acpi_ops_dflt = {
 	.enable_dma = sdhci_acpi_enable_dma,
 };
@@ -140,6 +146,7 @@ static const struct sdhci_ops sdhci_acpi_ops_int = {
 static const struct sdhci_ops sdhci_acpi_ops_int_cgm_emmc = {
 	.enable_dma = sdhci_acpi_enable_dma,
 	.hw_reset   = sdhci_acpi_int_hw_reset,
+	.get_min_clock = sdhci_acpi_cgm_get_min_clock,
 #ifdef CONFIG_HW_MUTEXES
 	.write_l = cgm_write_l,
 	.write_w = cgm_write_w,
-- 
1.7.9.5

